### PR TITLE
Orbit angular

### PR DESCRIPTION
**Pull request by JAndringa**
### Comments for the reviewer
The previous orbitAngular skill was not working and would just kick the ball immediately without turning. Also the code used a lot of magic numbers (those are evil) and I could not understand every line. So I decided to simply rewrite it, so that I know for sure that the math is correct. The robot now orbits around the center of the ball and then returns success when it is within the angle error margin. In my opinion the kicking mechanism should not be included because you don't want to kick while orbiting anyway, as this will cause the ball to curve (would be interesting to somehow use this to our advantage and confuse the enemy). The skill now uses only 1 "magic" number, which is a constraint from control on the angular velocity of the robot. The orbit works on both the simulation and in real life, though the simulation is working way better (I blame control).

###### Code Quality
- [x] Is the code is understandable and easy to read
- [x] Changes to the code comply with set clang-format rules
- [x] No use of manual memory control (e.g new/malloc/colloc etc)
- [x] Are (only) smart pointers used?

###### Testing
- [x] All tests are passing.
- [ ] I _added new / changed existing_ tests to reflect code changes (state why not otherwise!)
- [x] I tested my changes manually (Describe how, to what extent etc.)

###### Commit Messages
- [x] Commit message is saying what has been changed, **why** it was changed? Remember other developers might not know
  what the problem you are fixing was. Note also negative _decision_ (e.g., why did you not do particular thing)
  **TLDR: Commit message are comprehensive**
- [x] Commit messages follows the rules of https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by JAndringa** [comment]
btw the checks are not passing because the action is not working correctly this is fixed in another PR #37

File: roboteam_ai/src/stp/skills/OrbitAngular.cpp [hunk 1]
```diff
@@ -1,66 +1,40 @@
...
-    constexpr double ANGLE_RATE_FACTOR = 1.0;
-    constexpr double VELOCITY_FACTOR = 1.2;
-
-    // Calculate angVel and rotational velocity to make a circular movement around the ball
-    double angVel = -rpm / 60 * 2 * M_PI * ANGLE_RATE_FACTOR;
-    double rotVel = rpm / 60 * circumference * VELOCITY_FACTOR;
-
-    Vector2 directionVector = info.getRobot()->get()->getAngle().toVector2();
-    Angle targetAngle = (info.getPositionToShootAt().value() - info.getRobot()->get()->getPos()).toAngle();
+    RTT_DEBUG(info.getBall().value()->velocity.length())
```
Comment (Florisheinen1) [inline roboteam_ai/src/stp/skills/OrbitAngular.cpp:9]: Perhaps it is better to not leave this print statement in the code. If these print statements happen more often, eventually the terminal will get cluttered by so many statements that the few important statements are difficult to notice.

If you do want to keep this statement in here, I suggest adding some text around it so when it prints, on the same line it explains what the value means, not just a random number like 1.591 :)

**Comment by Florisheinen1** [comment]
Overall, the code looks fine. This is however one of those things of which the code might be mathematically correct, but practically unusable. Make sure to test this not just on the simulator, but also with the actual robot(s). If this works (better) than what was implemented before, I would accept this PR.

File: roboteam_ai/src/stp/skills/OrbitAngular.cpp (hunk 1 above)
Comment (JAndringa) [reply roboteam_ai/src/stp/skills/OrbitAngular.cpp]: It was indeed not the idea to leave this in the code. Will get rid of it.

**Review by umer99roboteam — APPROVED** [review]
LGTM!
Witnessed testing on the field and the velocity or orbit was tuned appropriately.
I see that the change from clockwise to counterclockwise for theta is in line with the changes made to the axis in control.

**Pull request merged**